src/app/modules/submission/submission.errors.ts

```diff
@@ -44,6 +44,16 @@ export class ValidateFieldError extends ApplicationError {
   }
 }
 
+/**
+ * A custom error class returned when a method explicitly returns a list of errors
+ * but the list itself is empty.
+ */
+export class EmptyErrorFieldError extends ApplicationError {
```

Comment:
this seems like a pretty general error type, so you can consider putting it in `core.errors.ts` so it can be shared for similar use cases in future

Reply:
👌  moved!